tests/test_mantatail.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+import socket
+import threading
+from mantatail import Server
+
+motd_dict_test = {
+    "motd": [
+        "- Hello {user_nick}, this is a test MOTD!",
+        "-",
+        "- Foo",
+        "- Bar",
+        "- Baz",
+        "-",
+        "- End test MOTD",
+    ]
+}
+
+##############
+#  FIXTURES  #
+##############
+
+# Based on: https://gist.github.com/sbrugman/59b3535ebcd5aa0e2598293cfa58b6ab#gistcomment-3795790
+@pytest.fixture(scope="function")
+def fail_test_if_there_is_an_error_in_a_thread(monkeypatch):
+    last_exception = None
+
+    class ThreadWrapper(threading.Thread):
+        def run(self):
+            try:
+                super().run()
+            except Exception as e:
+                nonlocal last_exception
+                last_exception = e
+
+    monkeypatch.setattr(threading, "Thread", ThreadWrapper)
+    yield
+    if last_exception:
+        raise last_exception
+
+
+@pytest.fixture(autouse=True, scope="function")
```

Comment:
`scope="function"` is the default, no need to change:

```suggestion
@pytest.fixture(autouse=True)
```